pkg/crinit/aggregated/aggregated_test.go

```diff
@@ -0,0 +1,152 @@
+			buffer: &bytes.Buffer{},
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test1",
+					ClusterRegistryNamespace: "cr-namespace",
+					DryRun: false,
+				},
+			},
+			client:      fake.NewSimpleClientset(),
+			objExpected: true,
+		},
+		{
+			desc:   "dry run should not create RBAC options",
+			buffer: &bytes.Buffer{},
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test2",
+					ClusterRegistryNamespace: "dry-namespace",
+					DryRun: true,
+				},
+			},
+			client:      fake.NewSimpleClientset(),
+			objExpected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			rbacObj, err := createRBACObjects(tc.buffer, tc.client, tc.opts)
+			if err != nil {
+				t.Error("Failed to create an RBAC Object")
+			}
+			chkrbac, _ := tc.client.CoreV1().ServiceAccounts(tc.opts.ClusterRegistryNamespace).Get(rbacObj.Name, metav1.GetOptions{})
+			if tc.objExpected {
+				if chkrbac == nil {
+					t.Errorf("Failed to create RBAC Object")
+				}
+			} else {
+				if chkrbac != nil {
+					t.Errorf("Expected no RBAC Object but got: %v", chkrbac)
+				}
+			}
+		})
+	}
+}
+
+func TestCreateAPIService(t *testing.T) {
+	testCases := []struct {
+		desc         string
+		buffer       *bytes.Buffer
+		opts         *aggregatedClusterRegistryOptions
+		svcClient    client.Interface
+		apisvcClient *fakeclientset.Clientset
+		svcExpected  bool
+	}{
+		{
+			desc:   "should create an API Service",
+			buffer: &bytes.Buffer{},
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test1",
+					ClusterRegistryNamespace: "cr-namespace",
+					DryRun: false,
+				},
+			},
+			svcClient:    fake.NewSimpleClientset(),
+			apisvcClient: fakeclientset.NewSimpleClientset(),
+			svcExpected:  true,
+		},
+		{
+			desc:   "should create an API Service",
+			buffer: &bytes.Buffer{},
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test2",
+					ClusterRegistryNamespace: "dry-namespace",
+					DryRun: true,
+				},
+			},
+			svcClient:    fake.NewSimpleClientset(),
+			apisvcClient: fakeclientset.NewSimpleClientset(),
+			svcExpected:  false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			svc, ips, hostnames, err := common.CreateService(tc.buffer, tc.svcClient, tc.opts.ClusterRegistryNamespace, tc.opts.Name, "", nil, v1.ServiceTypeClusterIP, false)
+			if err != nil {
+				t.Error("Error in pre-test creating service")
+			}
+			credentials, err := util.GenerateCredentials(tc.opts.ClusterRegistryNamespace, tc.opts.Name, svc.Name, "", ips, hostnames, false, false)
+			if err != nil {
+				t.Error("Error in pre-test generating credentials: %v", err)
+			}
+			apisvcObj, err := createAPIService(tc.buffer, tc.apisvcClient, tc.opts, util.GetCAKeyPair(credentials).Cert)
```

Comment:
It might make this test a bit more robust if you create these dependencies in the test code, rather than relying on `crinit`'s code to create the service and the credentials.